biz.aQute.bndlib/src/aQute/bnd/osgi/repository/SimpleIndexer.java

```diff
@@ -107,26 +122,24 @@ public static void index(Collection<File> files, OutputStream outputStream, URI
 		Objects.requireNonNull(outputStream, "'outputStream' argument cannot be null");
 		Objects.requireNonNull(base, "'base' argument cannot be null");
 
-		ResourcesRepository resourcesRepository = new ResourcesRepository();
-
-		files.stream()
+		ResourcesRepository resourcesRepository = files.stream()
 			.filter(f -> f.exists() && !f.isDirectory() && !f.isHidden() && f.canRead())
-			.forEach(f -> {
-				URI uri = base.relativize(f.toURI());
-
+			.map(f -> {
 				try {
 					ResourceBuilder resourceBuilder = new ResourceBuilder();
-					if (resourceBuilder.addFile(f, uri)) {
+					if (resourceBuilder.addFile(f, base.relativize(f.toURI()))) {
 						if (analyzer != null) {
-							analyzer.analyzeFile(f, resourceBuilder.getCapabilities(),
-								resourceBuilder.getRequirements());
+							analyzer.analyzeFile(f, new SafeResourceBuilder(resourceBuilder));
 						}
-						resourcesRepository.add(resourceBuilder.build());
+						return resourceBuilder.build();
 					}
 				} catch (Exception e) {
 					logger.error("Could not index file {}", f, e);
 				}
-			});
+				return null;
+			})
+			.filter(r -> Objects.nonNull(r))
```

Comment:
You do not need to filter for null. ResourcesRepository ignores adding null. :-)

https://github.com/bndtools/bnd/blob/eda1aa4fa34b785baea0cb5683d4b2d163b6c2b0/biz.aQute.bndlib/src/aQute/bnd/osgi/repository/ResourcesRepository.java#L48-L52